packages/rpc-graphql/src/loaders/account.ts

```diff
@@ -1,45 +1,334 @@
+                    }
+                    // Either the fetch arg set has no `dataSlice` argument, or
+                    // it couldn't be combined with another fetch, likely due to
+                    // the wasted byte limitation.
+                    // Add the fetch to the list as a new fetch.
+                    const argsHash = hashOmit(args, []);
+                    const accountFetches = (accountFetchesByArgsHash[argsHash] ||= {
+                        addresses: {},
+                        args,
+                    });
+                    const { callbacks: promiseCallbacksForAddress } = (accountFetches.addresses[address] ||= {
+                        callbacks: [],
+                    });
+                    promiseCallbacksForAddress.push({ callback: promiseCallback, dataSlice: args.dataSlice ?? null });
+                });
+            });
+
+            // Place the orphans
+            Object.entries(orphanedFetches).forEach(([address, toFetch]) => {
+                toFetch.forEach(({ args: orphanedArgs, promiseCallback: orphanPromiseCallback }) => {
+                    if (Object.keys(accountFetchesByArgsHash).length !== 0) {
+                        for (const { addresses, args } of Object.values(accountFetchesByArgsHash)) {
+                            // Check if the two arg sets are a match without
+                            // `encoding` and `dataSlice` arguments.
+                            if (
+                                hashOmit(orphanedArgs, ['encoding', 'dataSlice']) ===
+                                hashOmit(args, ['encoding', 'dataSlice'])
+                            ) {
+                                const { callbacks: promiseCallbacksForAddress } = (addresses[address] ||= {
+                                    callbacks: [],
+                                });
+                                promiseCallbacksForAddress.push({ callback: orphanPromiseCallback });
+                                return;
+                            }
+                        }
+                    }
+                    // Create a new fetch. Prefer `base64` encoding.
+                    const args: AccountLoaderArgsBase = { ...orphanedArgs, encoding: 'base64' };
+                    const argsHash = hashOmit(args, []);
+                    const accountFetches = (accountFetchesByArgsHash[argsHash] ||= {
+                        addresses: {},
+                        args,
+                    });
+                    const { callbacks: promiseCallbacksForAddress } = (accountFetches.addresses[address] ||= {
+                        callbacks: [],
+                    });
+                    promiseCallbacksForAddress.push({ callback: orphanPromiseCallback });
+                });
+            });
+
+            /**
+             * For each set of accounts related to some common args, fetch them in the fewest number
+             * of network requests.
+             */
+            await Promise.all(
```

Comment:
FWIW I don't think my [original take](https://github.com/solana-labs/solana-web3.js/pull/2063/files) had `await` in the `finally` block and yours doesn't have to either. All you want to do is make sure all the requests are fired off. You don't actually need to `await Promise.all()` from the perspective of `createAccountBatchLoadFn()()`, which has already returned.